packages/runner/lib/sdk/locks.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/require-await */
+import type { Result } from '@nangohq/utils';
+import { Err, Ok } from '@nangohq/utils';
+
+interface Lock {
+    key: string;
+    owner: string;
+    expiresAt: Date;
+}
+
+export class Locks {
+    private store = new Map<string, Lock>();
+    private mutex = new Map<string, boolean>();
+
+    public async tryAcquireLock({ owner, key, ttlMs }: { owner: string; key: string; ttlMs: number }): Promise<Result<boolean>> {
```

Comment:
those functions doesn't require async at the moment but in v2 the store will be a db so we will need it then.